temporalio/worker/workflow_instance.py

```diff
@@ -0,0 +1,1396 @@
+    ) -> None:
+        # No matter the result, we know we want to pop
+        handle = self._pending_child_workflows.pop(job.seq, None)
+        if not handle:
+            raise RuntimeError(
+                f"Failed finding child workflow handle for sequence {job.seq}"
+            )
+        if job.result.HasField("completed"):
+            ret: Optional[Any] = None
+            if job.result.completed.HasField("result"):
+                ret_types = [handle._input.ret_type] if handle._input.ret_type else None
+                ret_vals = self._convert_payloads(
+                    [job.result.completed.result],
+                    ret_types,
+                )
+                ret = ret_vals[0]
+            handle._resolve_success(ret)
+        elif job.result.HasField("failed"):
+            handle._resolve_failure(
+                temporalio.exceptions.failure_to_error(
+                    job.result.failed.failure, self._payload_converter
+                )
+            )
+        elif job.result.HasField("cancelled"):
+            handle._resolve_failure(
+                temporalio.exceptions.failure_to_error(
+                    job.result.cancelled.failure, self._payload_converter
+                )
+            )
+        else:
+            raise RuntimeError("Child workflow did not have result")
+
+    def _apply_resolve_child_workflow_execution_start(
+        self,
+        job: temporalio.bridge.proto.workflow_activation.ResolveChildWorkflowExecutionStart,
+    ) -> None:
+        handle = self._pending_child_workflows.get(job.seq)
+        if not handle:
+            raise RuntimeError(
+                f"Failed finding child workflow handle for sequence {job.seq}"
+            )
+        if job.HasField("succeeded"):
+            handle._resolve_start_success(job.succeeded.run_id)
+        elif job.HasField("failed"):
+            self._pending_child_workflows.pop(job.seq)
+            if (
+                job.failed.cause
+                == temporalio.bridge.proto.child_workflow.StartChildWorkflowExecutionFailedCause.START_CHILD_WORKFLOW_EXECUTION_FAILED_CAUSE_WORKFLOW_ALREADY_EXISTS
+            ):
+                handle._resolve_failure(
+                    temporalio.exceptions.WorkflowAlreadyStartedError(
+                        job.failed.workflow_id, job.failed.workflow_type
+                    )
+                )
+            else:
+                handle._resolve_failure(
+                    RuntimeError("Unknown child start fail cause: {job.failed.cause}")
+                )
+        elif job.HasField("cancelled"):
+            self._pending_child_workflows.pop(job.seq)
+            handle._resolve_failure(
+                temporalio.exceptions.failure_to_error(
+                    job.cancelled.failure, self._payload_converter
+                )
+            )
+        else:
+            raise RuntimeError("Child workflow start did not have status")
```

Comment:
nit: Technically it's not a "known" status or no status.

Reply:
Ok

Reply:
Updated